blindsign/blindrsa/brsa.go

```diff
@@ -28,271 +26,176 @@ import (
 	"crypto"
 	"crypto/rand"
 	"crypto/rsa"
-	"hash"
 	"io"
 	"math/big"
 
 	"github.com/cloudflare/circl/blindsign/blindrsa/internal/common"
 	"github.com/cloudflare/circl/blindsign/blindrsa/internal/keys"
 )
 
-// An randomBRSAVerifier represents a Verifier in the RSA blind signature protocol.
-// It carries state needed to produce and validate an RSA signature produced
-// using the blind RSA protocol.
-type randomBRSAVerifier struct {
-	// Public key of the Signer
-	pk *rsa.PublicKey
-
-	// Identifier of the cryptographic hash function used in producing the message signature
-	cryptoHash crypto.Hash
-
-	// Hash function used in producing the message signature
-	hash hash.Hash
-}
+type Variant int
 
-// A deterministicBRSAVerifier is a BRSAVerifier that supports deterministic signatures.
-type deterministicBRSAVerifier struct {
-	// Public key of the Signer
-	pk *rsa.PublicKey
-
-	// Identifier of the cryptographic hash function used in producing the message signature
-	cryptoHash crypto.Hash
+const (
+	SHA384PSSRandomized        Variant = iota // RSABSSA-SHA384_PSS_Randomized
+	SHA384PSSZeroRandomized                   // RSABSSA-SHA384_PSSZero_Randomized
+	SHA384PSSDeterministic                    // RSABSSA-SHA384_PSS_Deterministic
+	SHA384PSSZeroDeterministic                // RSABSSA-SHA384_PSSZero_Deterministic
+)
 
-	// Hash function used in producing the message signature
-	hash hash.Hash
+func (v Variant) String() string {
+	switch v {
+	case SHA384PSSRandomized:
+		return "RSABSSA-SHA384-PSS-Randomized"
+	case SHA384PSSZeroRandomized:
+		return "RSABSSA-SHA384-PSSZero-Randomized"
+	case SHA384PSSDeterministic:
+		return "RSABSSA-SHA384-PSS-Deterministic"
+	case SHA384PSSZeroDeterministic:
+		return "RSABSSA-SHA384-PSSZero-Deterministic"
+	default:
+		panic(ErrInvalidVariant)
```

Comment:
`panic` sounds brutal for calling `String`. I understand `Stringer` does now allow to return an error, however, what about `return "invalid RSABSSA"` instead?

Reply:
done